### PR TITLE
test

### PR DESCRIPTION
**Pull request by redox0018**

### CODE REVIEW

File: src/main/java/Test.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        // Output details using slf4j
+        log.info("Monthly Salary: {}", monthlySalary);
+        log.info("Tax Rate: {}%", (taxRate * 100));
+        log.info("Annual Tax Paid: {}", annualTax);
+        log.info("Net Annual Salary: {}", netSalary);
+    }
+
+    private static double getTaxRate(double monthlySalary) {
+        if (monthlySalary <= 2500) {
+            return 0.05; // 5% tax rate
```
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:23]: Potential null pointer on field access '0' - consider adding null check

File: src/main/java/Test.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class Test {
+    public void calculateFunction(Long principal) {
```
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:5]: Method parameter 'principal' might be null - consider adding null check

File: src/main/java/Test.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        log.info("Tax Rate: {}%", (taxRate * 100));
+        log.info("Annual Tax Paid: {}", annualTax);
+        log.info("Net Annual Salary: {}", netSalary);
+    }
+
+    private static double getTaxRate(double monthlySalary) {
+        if (monthlySalary <= 2500) {
+            return 0.05; // 5% tax rate
+        } else if (monthlySalary <= 5000) {
+            return 0.10; // 10% tax rate
```
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:25]: Potential null pointer on field access '0' - consider adding null check

File: src/main/java/Test.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        } else if (monthlySalary <= 10000) {
+            return 0.20; // 20% tax rate
+        } else {
+            return 0.30; // 30% tax rate
```
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:29]: Returning potentially null value '0' - consider adding null check

File: src/main/java/Test.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        log.info("Net Annual Salary: {}", netSalary);
+    }
+
+    private static double getTaxRate(double monthlySalary) {
+        if (monthlySalary <= 2500) {
+            return 0.05; // 5% tax rate
+        } else if (monthlySalary <= 5000) {
+            return 0.10; // 10% tax rate
+        } else if (monthlySalary <= 10000) {
+            return 0.20; // 20% tax rate
```
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:27]: Potential null pointer on field access '0' - consider adding null check

File: src/main/java/Test.java (hunk 4 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:29]: Potential null pointer on field access '0' - consider adding null check

File: src/main/java/Test.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        // Calculate monthly salary
+        double monthlySalary = principal / 12;
+
+        // Calculate tax slab and net tax paid
+        double taxRate = getTaxRate(monthlySalary);
+        double annualTax = principal * taxRate;
+        double netSalary = principal - annualTax;
+
+        // Output details using slf4j
+        log.info("Monthly Salary: {}", monthlySalary);
```
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:15]: Potential null pointer on method call 'log' - consider adding null check

File: src/main/java/Test.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        log.info("Tax Rate: {}%", (taxRate * 100));
```
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:16]: Potential null pointer on method call 'log' - consider adding null check

File: src/main/java/Test.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        log.info("Annual Tax Paid: {}", annualTax);
```
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:17]: Potential null pointer on method call 'log' - consider adding null check

File: src/main/java/Test.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        log.info("Net Annual Salary: {}", netSalary);
```
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:18]: Potential null pointer on method call 'log' - consider adding null check

File: src/main/java/Test.java (hunk 5 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:27]: Returning potentially null value '0' - consider adding null check

File: src/main/java/Test.java (hunk 3 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:25]: Returning potentially null value '0' - consider adding null check

File: src/main/java/Test.java (hunk 1 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:23]: Returning potentially null value '0' - consider adding null check

**Review by piramal-codeReviewer — COMMENTED** [review]
Automated code review completed with 14 new comments.
Please address the issues found.

File: src/main/java/Test.java (hunk 2 above)
Comment (redox0018) [reply src/main/java/Test.java:5]: resolved

File: src/main/java/Test.java (hunk 6 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:15]: Potential null pointer on method call 'log' - consider adding null check

File: src/main/java/Test.java (hunk 1 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:23]: Potential null pointer on field access '0' - consider adding null check

File: src/main/java/Test.java (hunk 7 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:16]: Potential null pointer on method call 'log' - consider adding null check

File: src/main/java/Test.java (hunk 8 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:17]: Potential null pointer on method call 'log' - consider adding null check

File: src/main/java/Test.java (hunk 3 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:25]: Potential null pointer on field access '0' - consider adding null check

File: src/main/java/Test.java (hunk 9 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:18]: Potential null pointer on method call 'log' - consider adding null check

File: src/main/java/Test.java (hunk 5 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:27]: Potential null pointer on field access '0' - consider adding null check

File: src/main/java/Test.java (hunk 4 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:29]: Potential null pointer on field access '0' - consider adding null check

**Review by piramal-codeReviewer — COMMENTED** [review]
Automated code review completed with 9 new comments.
Please address the issues found.

File: src/main/java/Test.java (hunk 1 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:23]: Potential null pointer on field access '0' - consider adding null check

File: src/main/java/Test.java (hunk 3 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:25]: Potential null pointer on field access '0' - consider adding null check

File: src/main/java/Test.java (hunk 5 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:27]: Potential null pointer on field access '0' - consider adding null check

File: src/main/java/Test.java (hunk 4 above)
Comment (piramal-codeReviewer) [inline src/main/java/Test.java:29]: Potential null pointer on field access '0' - consider adding null check

**Review by piramal-codeReviewer — COMMENTED** [review]
Automated code review completed with 4 new comments.
Please address the issues found.

**Pull request merged**